### PR TITLE
[data] feat: add byted loader backend

### PR DESCRIPTION
**Pull request by YLlllllllllll**
## Summary

- Add an optional `byted_loader` backend for `data.dataloader.type`.
- Add a dataset bypass path so byted loader owns storage reads, scheduling, shuffle, global-batch construction, and progress state while VeOmni keeps transforms and `MainCollator`.
- Add a compatibility adapter with lazy `bytedance.dataloader` import, API contract checks, `state_dict` / `load_state_dict`, metadata sanitization, and effective-config logging.
- Document the v1 boundary: local mode, Lance/JSONL primary path, restricted Parquet compatibility, SP handled by `MainCollator`, PP/TP not claimed.

## Tests

- `git diff --cached --check`
- `python -m py_compile veomni/data/byted_loader.py veomni/data/data_loader.py veomni/data/__init__.py veomni/arguments/arguments_types.py veomni/trainer/base.py tests/data/test_byted_loader.py`
- `uv run --group lint ruff check tasks tests veomni docs`
- `uv run --group lint ruff format --check tasks tests veomni docs`
- `PYTHONNOUSERSITE=1 PYTHONPATH=/home/tiger/Open-VeOmni-github:/home/tiger/Open-VeOmni-github/.venv/lib/python3.11/site-packages python -m pytest /home/tiger/Open-VeOmni-github/tests/data/test_byted_loader.py -q`

## Notes

The real `bytedance.dataloader` package is imported only when `data.dataloader.type=byted_loader` is selected. Native dataloader paths do not require that optional package.

### CODE REVIEW

**Comment by YLlllllllllll** [comment]
Closing this PR because the external dependency (`bytedance.dataloader`) is not an open-source component and should not be wired directly into the public Open-VeOmni codebase.

The intended integration path is now to keep Open-VeOmni clean and register the backend from an internal VeOmni/ModelChef patch layer. The adapter and validation work will continue there, with `data.dataloader.type=byted_loader` remaining an internal optional backend rather than an upstream public feature.
